sentry-android-replay/src/main/java/io/sentry/android/replay/ReplayIntegration.kt

```diff
@@ -259,4 +256,67 @@ public class ReplayIntegration(
     override fun onTouchEvent(event: MotionEvent) {
         captureStrategy?.onTouchEvent(event)
     }
+
+    private fun cleanupReplays(unfinishedReplayId: String = "") {
+        // clean up old replays
+        options.cacheDirPath?.let { cacheDir ->
+            File(cacheDir).listFiles { dir, name ->
+                if (name.startsWith("replay_") &&
+                    !name.contains(replayId.toString()) &&
+                    !(unfinishedReplayId.isNotBlank() && name.contains(unfinishedReplayId))
+                ) {
+                    FileUtils.deleteRecursively(File(dir, name))
+                }
+                false
```

Comment:
I don't think the `.listFiles(filter: FileNameFilter)` ([docs here](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#listFiles--)) is meant to be used this way, as your filter code deletes folders while it is still iterating 😅 How about this?
```suggestion
            File(cacheDir).listFiles()?.forEach { file ->
                val name = file.name
                if (name.startsWith("replay_") &&
                    !name.contains(replayId.toString()) &&
                    !(unfinishedReplayId.isNotBlank() && name.contains(unfinishedReplayId))
                ) {
                    FileUtils.deleteRecursively(file)
                }
            }
```

Reply:
yeah, I've figured this in some other place but forgot to change here 😅 thanks for pointing out!